datafiles/manager.py

```diff
@@ -33,31 +34,30 @@ def all(self) -> Iterator[HasDatafile]:
         for filename in iglob(splatted):
             log.debug(f'Found matching path: {filename}')
             results = parse(pattern, filename)
-            fields = dataclasses.fields(self.model)
-            args = list(results.named.values())
-            args += [Missing] * (len(fields) - len(args))
-            yield self.model(*args)
+            yield self.get(*results.named.values())
 
-    def get_or_none(self, *args, **kwargs) -> Optional[HasDatafile]:
-        original_manual = self.model.Meta.datafile_manual
+    def get(self, *args, **kwargs) -> HasDatafile:
+        fields = dataclasses.fields(self.model)
+        args = list(args)
+        args += [Missing] * (len(fields) - len(args) - len(kwargs))
 
-        self.model.Meta.datafile_manual = True
-        instance = self.model(*args, **kwargs)
-        self.model.Meta.datafile_manual = original_manual
+        with hooks.disabled():
+            instance = self.model(*args, **kwargs)
+            instance.datafile.load()
 
-        if instance.datafile.exists:
-            instance.datafile._manual = original_manual
-            return instance
+        return instance
 
-        return None
+    def get_or_none(self, *args, **kwargs) -> Optional[HasDatafile]:
+        try:
+            return self.get(*args, **kwargs)
+        except FileNotFoundError:
+            return None
 
     def get_or_create(self, *args, **kwargs) -> HasDatafile:
-        instance = self.model(*args, **kwargs)
-
-        if not instance.datafile.exists:
-            instance.datafile.save()
```

Comment:
The CI is failing because I'm not doing this anymore. Any hint on why it's needed to manually save? Should this be triggered automatically?

Reply:
It's possible for the model to be defined with `manual=True` (https://datafiles.readthedocs.io/en/latest/api/model/) to disable automatic loading and saving, but the assumption is that after calling `get_or_create()` the file should always exist.

Reply:
But, mhh... I don't see any `manual=True` for the model used in the test case, where is it?

> the assumption is that after calling get_or_create() the file should always exist.

Why would I want the file created automatically if I explicitly asked to handle things manually?

Reply:
My inspiration here was the Django ORM, where `MyModel(...)` doesn't touch the database, but `MyModel.objects.create(...)` does.

Reply:
But there isn't two different modes in Django, or is there? Genuinely asking, I don't know much about Django!

Anyway, I'm most likely going to add that manual `save` so that I don't add a breaking change on an unrelated PR, but I still do not understand why the test doesn't pass when I don't see a `manual=True`. Can you help me?

Reply:
Alright, found the problem. See #146!